### PR TITLE
Feature/feature update pet

### PR DESCRIPTION
**Pull request by enrichp**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 19
           

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Gender.java  100
         

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetRepeatException.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePet.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990&bid=17115001)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    @Test
+    public void shouldChangeGender() {
+        pet.setGender(Gender.MALE);
+        assertEquals(Gender.MALE, pet.getGender());
+    }
+
+    @Test
+    public void shouldChangeWeight() {
+        pet.setWeight(30);
+        assertEquals(30, pet.getWeight(),0);
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: [',' is not followed by whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    @Test
+    public void shouldChangeGender() {
+        pet.setGender(Gender.MALE);
+        assertEquals(Gender.MALE, pet.getGender());
+    }
+
+    @Test
+    public void shouldChangeWeight() {
+        pet.setWeight(30);
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: ['30' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        User usr2 = new User ( "Gabi", "er2@gmail.com", "909020");
+        usr2.addPet(getAuxPet("Manolo"));
+        trUpdatePet.setUser(usr2);
+        trUpdatePet.setPet(pet);
+        trUpdatePet.execute();
+    }
+
+    @Test(expected = PetRepeatException.class)
+    public void shouldNotUpdatePetNameIfExisting() throws PetRepeatException {
+        Pet linux = getAuxPet("Linux");
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: [The String "Linux" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    @Test
+    public void shouldChangeBreed() {
+        pet.setBreed("Bulldog");
+        assertEquals("Bulldog", pet.getBreed());
+    }
+
+    @Test
+    public void shouldUpdatePetService() throws UserIsNotOwnerException {
+        usr.addPet(pet);
+        pet.setWeight(20);
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: ['20' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePet.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
+package org.pesmypetcare.mypetcare.controllers;
+
+import org.pesmypetcare.mypetcare.features.pets.Pet;
+import org.pesmypetcare.mypetcare.features.pets.PetRepeatException;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePet.java]: Codacy Issue found: [Unused import - org.pesmypetcare.mypetcare.features.pets.PetRepeatException.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+    @Test
+    public void shouldChangeName() throws PetRepeatException {
+        pet.setName("Ohio");
+        assertEquals("Ohio", pet.getName());
+    }
+
+    @Test
+    public void shouldChangeBreed() {
+        pet.setBreed("Bulldog");
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: [The String "Bulldog" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    private TrUpdatePet trUpdatePet;
+
+    @Before
+    public void setUp() throws PetRepeatException {
+        usr = new User("johnDoe", "johndoe@gmail.com", "1234");
+        pet = new Pet();
+        pet.setName("Manolo");
+        pet.setGender(Gender.FEMALE);
+        pet.setBirthDate("2 MAR 2010");
+        pet.setBreed("Husky");
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: [The String "Husky" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetRepeatException.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package org.pesmypetcare.mypetcare.features.pets;
+
+public class PetRepeatException extends Exception {
+    public PetRepeatException() {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetRepeatException.java]: Codacy Issue found: [Avoid unnecessary constructors - the compiler will generate these for you](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java (hunk 1 above)
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: ['30' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePet.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+     */
+    private boolean userIsNotTheOwnerOfThePet() {
+        return !pet.getOwner().equals(user);
+    }
+
+    /**
+     * Get the result of the transaction.
+     * @return The result of the transaction
+     */
+    public boolean getResult() {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePet.java]: Codacy Issue found: [A getX() method which returns a boolean should be named isX()](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+public class TestPetUpdate {
+    private Pet pet;
+    private User usr;
+    private TrUpdatePet trUpdatePet;
+
+    @Before
+    public void setUp() throws PetRepeatException {
+        usr = new User("johnDoe", "johndoe@gmail.com", "1234");
+        pet = new Pet();
+        pet.setName("Manolo");
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: [The String "Manolo" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+        boolean changeResult = trUpdatePet.getResult();
+
+        assertTrue("should communicate with service to update a pet", changeResult);
+
+    }
+
+    @Test(expected = UserIsNotOwnerException.class)
+    public void shouldNotUpdatePetIfNotOwner() throws PetRepeatException, UserIsNotOwnerException {
+        usr.addPet(pet);
+        User usr2 = new User ( "Gabi", "er2@gmail.com", "909020");
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: ['(' is preceded with whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+    @Test
+    public void shouldChangeWeight() {
+        pet.setWeight(30);
+        assertEquals(30, pet.getWeight(),0);
+    }
+
+    @Test
+    public void shouldChangeName() throws PetRepeatException {
+        pet.setName("Ohio");
+        assertEquals("Ohio", pet.getName());
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: [JUnit assertions should include a message](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java (hunk 11 above)
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: ['(' is followed by whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+        pet.setRecommendedDailyKiloCalories(2);
+        pet.setWashFrequency(2);
+        pet.setWeight(2);
+        trUpdatePet = new TrUpdatePet(new StubPetManagerService());
+    }
+
+    @Test
+    public void shouldChangeGender() {
+        pet.setGender(Gender.MALE);
+        assertEquals(Gender.MALE, pet.getGender());
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: [JUnit assertions should include a message](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+    @Test
+    public void shouldChangeName() throws PetRepeatException {
+        pet.setName("Ohio");
+        assertEquals("Ohio", pet.getName());
+    }
+
+    @Test
+    public void shouldChangeBreed() {
+        pet.setBreed("Bulldog");
+        assertEquals("Bulldog", pet.getBreed());
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: [JUnit assertions should include a message](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetRepeatException.java [hunk 15]
```diff
@@ -0,0 +1,7 @@
...
+        super();
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/features/pets/PetRepeatException.java]: Codacy Issue found: [Unnecessary call to superclass constructor with no arguments.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+
+    @Test
+    public void shouldChangeWeight() {
+        pet.setWeight(30);
+        assertEquals(30, pet.getWeight(),0);
+    }
+
+    @Test
+    public void shouldChangeName() throws PetRepeatException {
+        pet.setName("Ohio");
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: [The String "Ohio" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java (hunk 1 above)
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: [JUnit assertions should include a message](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Gender.java  100
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePet.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990&bid=17115001)

File: app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+
+    @Test
+    public void shouldChangeWeight() {
+        pet.setWeight(number30);
+        assertEquals("Correct weight change", number30, pet.getWeight(), 0);
+    }
+
+    @Test
+    public void shouldChangeName() throws PetRepeatException {
+        pet.setName("Ohio");
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java]: Codacy Issue found: [The String "Ohio" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Gender.java  100
         

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePet.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990&bid=17115001)

**Review by AlbertPG — APPROVED** [review]
I see no problem at all

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePet.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990&bid=17115001)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePet.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990&bid=17115001)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdate.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUpdatePet.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5219990&bid=17115001)

**Pull request merged**